src/aiida/engine/daemon/execmanager.py

```diff
@@ -360,15 +361,14 @@ def _copy_local_files(logger, node, transport, inputs, local_copy_list):
             if data_node.base.repository.get_object(filename_source).file_type == FileType.DIRECTORY:
                 # If the source object is a directory, we copy its entire contents
                 data_node.base.repository.copy_tree(filepath_target, filename_source)
+                transport.put(f'{dirpath}/*', target or '.')
             else:
                 # Otherwise, simply copy the file
                 with filepath_target.open('wb') as handle:
                     with data_node.base.repository.open(filename_source, 'rb') as source:
                         shutil.copyfileobj(source, handle)
-
-        # Now copy the contents of the temporary folder to the remote working directory using the transport
-        for filepath in dirpath.iterdir():
-            transport.put(str(filepath), filepath.name)
+                transport.makedirs(str(pathlib.Path(target).parent), ignore_existing=True)
```

Comment:
Just a note that because of line 359 and this one, the copy behaviour of `local_copy_list` is _not_ the same as `cp`, which would simply fail in case the parent folder that you are trying to copy into doesn't exist.

I wonder if this was also the previous behavior of `local_copy_list`. The QE plugin made the `pseudo` directory in the sandbox folder exactly because otherwise the copy command would fail, I assume.

Finally, `remote_copy_list` _does_ fail when trying to copy files into a parent folder that doesn't exist.